lib/rules/no-arrow-function-lifecycle.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @fileoverview It is not necessary to use arrow function for lifecycle methods
+ * @author Tan Nguyen
+ */
+'use strict';
+
+const Components = require('../util/Components');
+const astUtil = require('../util/ast');
+const docsUrl = require('../util/docsUrl');
+
+const lifecycleMethods = [
+  'getDefaultProps',
+  'getInitialState',
+  'getChildContext',
+  'getDerivedStateFromProps',
+  'componentWillMount',
+  'UNSAFE_componentWillMount',
+  'componentDidMount',
+  'componentWillReceiveProps',
+  'UNSAFE_componentWillReceiveProps',
+  'shouldComponentUpdate',
+  'componentWillUpdate',
+  'UNSAFE_componentWillUpdate',
+  'getSnapshotBeforeUpdate',
+  'componentDidUpdate',
+  'componentDidCatch',
+  'componentWillUnmount',
+  'render'
+];
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'It is not necessary to use arrow function for lifecycle methods',
+      category: 'Best Practices',
+      recommended: true,
```

Comment:
Just to confirm, do we want this to be recommended? that would be a semver-major change, right?

Reply:
ah yes, it would - let's set this to false.

Reply:
good call, let's set this to false